e2e/cypress/tests/integration/enterprise/self-hosted/self_hosted_purchase_spec.ts

```diff
@@ -0,0 +1,382 @@
+                    const filePath = Cypress.config('downloadsFolder') + '/' + fileName + '.pdf';
+                    cy.writeFile(filePath, response.body, 'binary');
+                    cy.task('getPdfContent', filePath).then((data) => {
+                        const allLines = (data as {text: string}).text.split('\n');
+                        const prodLine = allLines.filter((line) => line.includes('Self-Hosted Professional'));
+                        expect(prodLine.length).to.be.equal(1);
+                        getCurrentUsers().then((userCount) => {
+                            cy.wait('@products').then((res) => {
+                                const product = res.response.body.find((product: Cypress.Product) => product.sku === 'professional');
+                                const purchaseAmount = dollarUSLocale.format((userCount + additionalSeatsToPurchase) * (product).price_per_seat * 12);
+                                const amountLine = allLines.find((line: string) => line.includes('Amount paid'));
+                                if (!amountLine.includes(purchaseAmount)) {
+                                    throw new Error(`Expected purchase amount ${purchaseAmount}, but amount line was ${amountLine}`);
+                                }
+                            });
+                        });
+                    });
+                },
+            );
+        });
+
+        // * Check that creating groups, a professional feature, is now available for use.
+        cy.visit('/');
+        cy.uiGetProductMenuButton().click();
+        cy.contains('User Groups').click();
+        cy.contains('Create Group').should('be.enabled');
+    });
+
+    it('must purchase a license for at least the current number of users', () => {
+        cy.apiAdminLogin();
+        cy.visit('/');
+        cy.apiDeleteLicense();
+
+        cy.intercept('GET', '**/api/v4/hosted_customer/signup_available').as('airGappedCheck');
+        cy.intercept('GET', 'https://js.stripe.com/v3').as('stripeCheck');
+        cy.intercept('GET', '**/api/v4/cloud/products/selfhosted').as('products');
+
+        // # Open pricing modal
+        cy.get('#UpgradeButton').should('exist').click();
+
+        cy.wait('@airGappedCheck');
+        cy.wait('@stripeCheck');
+        cy.wait('@products');
+
+        // The waits for these fetches is usually enough. Add a little wait
+        // for all the selectors to be updated and rerenders to happen
+        // so that we do not accidentally hit the air-gapped modal
+        // eslint-disable-next-line cypress/no-unnecessary-waiting
+        cy.wait(50);
+
+        // # Click the upgrade button to open the modal
+        cy.get('#professional_action').should('exist').click();
+
+        // * Verify basic purchase elements are available
+        verifyPurchaseModal();
+
+        // # Fill form with too low of a number of seats
+        fillForm({...defaultSuccessForm, seats: 1}, getCurrentUsers());
```

Comment:
Curious, where do we add additional users besides the admin one?

Reply:
Lots of tests add a new user as they go along, and in local runs we don't know how many users there will already be. The `apiInitSetup` helper is probably the biggest creator of users in e2e tests: https://github.com/mattermost/mattermost-webapp/blob/master/e2e/cypress/tests/support/api/setup.ts#L22-L35